Dockerfile

```diff
@@ -1,5 +1,15 @@
-FROM golang:1.19.1-alpine3.16
-RUN mkdir -p /app/data
-WORKDIR /app
-RUN go build -o main .
-ENTRYPOINT ["/app/main"]
+FROM golang:1.20
+
+VOLUME /secret/data
+
+WORKDIR /usr/src/secret
+
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
+
+COPY . .
+RUN go build -v -o /usr/local/bin/secret awesomeProject3
```

Comment:
Remember that dockerfile should be build with multi-staged approach according to the specification.